src/vm/tests/integrations.rs

```diff
@@ -0,0 +1,229 @@
+use vm::{
+    database::HeadersDB,
+    types::QualifiedContractIdentifier,
+    Value, ClarityName, ContractName, errors::RuntimeErrorType, errors::Error as ClarityError };
+use chainstate::stacks::{
+    db::StacksChainState, C32_ADDRESS_VERSION_TESTNET_SINGLESIG,
+    StacksPrivateKey, TransactionSpendingCondition, TransactionAuth, TransactionVersion,
+    StacksPublicKey, TransactionPayload, StacksTransactionSigner,
+    StacksTransaction, TransactionSmartContract, TransactionContractCall, StacksAddress };
+use chainstate::burn::VRFSeed;
+use burnchains::Address;
+use address::AddressHashMode;
+use net::{Error as NetError, StacksMessageCodec};
+use util::{log, strings::StacksString, hash::hex_bytes, hash::to_hex};
+
+use util::db::{DBConn, FromRow};
+
+use testnet;
+use testnet::mem_pool::MemPool;
+
+fn serialize_sign_standard_single_sig_tx(payload: TransactionPayload,
+                                         sender: &StacksPrivateKey, nonce: u64) -> Vec<u8> {
+    let mut spending_condition = TransactionSpendingCondition::new_singlesig_p2pkh(StacksPublicKey::from_private(sender))
+        .expect("Failed to create p2pkh spending condition from public key.");
+    spending_condition.set_nonce(nonce);
+    spending_condition.set_fee_rate(0);
+    let auth = TransactionAuth::Standard(spending_condition);
+    let unsigned_tx = StacksTransaction::new(TransactionVersion::Testnet, auth, payload);
+    let mut tx_signer = StacksTransactionSigner::new(&unsigned_tx);
+    tx_signer.sign_origin(sender).unwrap();
+    tx_signer.get_tx().unwrap().serialize()    
+}
+
+fn make_contract_publish(sender: &StacksPrivateKey, nonce: u64, contract_name: &str, contract_content: &str) -> Vec<u8> {
+    let name = ContractName::from(contract_name);
+    let code_body = StacksString::from_string(&contract_content.to_string()).unwrap();
+
+    let payload = TransactionSmartContract { name, code_body };
+
+    serialize_sign_standard_single_sig_tx(payload.into(), sender, nonce)
+}
+
+fn make_contract_call(
+    sender: &StacksPrivateKey, nonce: u64,
+    contract_addr: &StacksAddress, contract_name: &str,
+    function_name: &str, function_args: &[Value]) -> Vec<u8> {
+
+    let contract_name = ContractName::from(contract_name);
+    let function_name = ClarityName::from(function_name);
+
+    let payload = TransactionContractCall {
+        address: contract_addr.clone(),
+        contract_name, function_name,
+        function_args: function_args.iter().map(|x| x.clone()).collect()
```

Comment:
Not addressed in this PR, but is there a maximum allowed length for the number of arguments to a Clarity function?  Or is it bound only by the compute budget?

Reply:
Currently only bound by the compute budget.

Reply:
Thanks!  I'll make a note of that in the transaction serialization and processing code as well.